### DISCUSSION TITLE
Architecture of Web APP

### DISCUSSION THREAD

**Discussion by Nanguage** [flowsaber/flowsaber#5] [General]
I have readed the Feishu project doc, I'm considering how to organize the relation of UI, server, and task runner. Here is my understanding:

I think the flow defination and running status should be distinguish. 
UI is for show/compose the flow defination and serialize it, pass it to server. Server recive the flow def and pass to runner to run, and provide a interface for UI fetch the running status.

![image](https://user-images.githubusercontent.com/14010939/113747326-d9ec3c80-9739-11eb-9b70-dde0899bc2e0.png)

It's that right?

**Comment by zqbake** [comment]
how about let the ruuner just communicate with the server? This could simplify the rw process
 of the database and also relieve the burden of the runner to maintain two connection channels at the same time.

**Reply by zqbake** [reply]
The graphql schema I showed in the FeiShu doc has both query and mutation operations for fetching and updating flow/tasks.

**Reply by Nanguage** [reply]
Modified:
![image](https://user-images.githubusercontent.com/14010939/114057800-7db62380-98c5-11eb-8251-af0b1cb46b07.png)

**Reply by zqbake** [reply]
current architecture in dev branch:

![image](https://user-images.githubusercontent.com/29366130/116442986-b4a8a500-a885-11eb-8da3-6139b25e6a30.png)
